### PR TITLE
(feat) react-component:  Working on example code and the course

### PR DESCRIPTION
**Pull request by nilscognite**
#### Type of change

Feat  

Description:

* Corrected all types using spell checker extension, in filenames, comments and names in general (a lot of files)
* Updated some comments 
* Add buttonType to BaseCommand and use it in the react code
* Added some convenience methods in the BaseEditTool
* Updated DomainObjectChange  to more flexible
* Add the command ShowExamplesOnTopCommand for the example code
* Remove the old toolbar in the architecture story
* Finished the README.md for the course

### CODE REVIEW

**Comment by haakonflatval-cognite** [comment]
Title should be `feat(react-components): ...`

File: react-components/src/components/Architecture/CommandButton.tsx [hunk 1]
```diff
@@ -67,10 +67,15 @@ export const CommandButton = ({
   }
   const placement = isHorizontal ? 'top' : 'right';
   const { key, fallback } = newCommand.tooltip;
+  // This was the only way it went through compiler: (more bytton types will be added in the future)
+  const type = newCommand.buttonType;
+  if (type !== 'ghost' && type !== 'ghost-destructive') {
+    return <></>;
+  }
```
Comment (haakonflatval-cognite) [inline react-components/src/components/Architecture/CommandButton.tsx:74]: Just curious - what kind of error do you get when you don't have this?

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM :+1:

File: react-components/src/components/Architecture/CommandButton.tsx (hunk 1 above)
Comment (nilscognite) [reply react-components/src/components/Architecture/CommandButton.tsx:74]: I try to cast, but ButtonType/GhostButtonType is not exposed. It has to be a ButtonType or a GhostButtonType. The syntax i cogs is really hard to understand.

**Pull request merged**